File: legacy/_1.7.10/src/main/java/cn/enaium/noexpensive/command/NoExpensiveCommand.java
package cn.enaium.noexpensive.command;

import cn.enaium.noexpensive.Config;
import net.minecraft.command.AbstractCommand;
import net.minecraft.command.CommandSource;
import net.minecraft.text.LiteralText;
import net.minecraft.text.TranslatableText;
import org.jetbrains.annotations.NotNull;

/**
 * @author Enaium
 */
public class NoExpensiveCommand extends AbstractCommand {

    @Override
    public String getCommandName() {
        return "noexpensive";
    }

    @Override
    public String getUsageTranslationKey(CommandSource source) {
        return null;
    }

    @Override
    public void execute(CommandSource source, String[] args) {
        if (args.length == 0) {
            source.sendMessage(new LiteralText("§cNoExpensive §7by §bEnaium"));
            return;
        }

        if (args[0].equalsIgnoreCase("reload")) {
            Config.load();
            source.sendMessage(new TranslatableText("command.reload.success"));
        }
    }

    @Override
    public int compareTo(@NotNull Object o) {
        return 0;
    }
}

File: legacy/_1.8.9/src/main/kotlin/cn/enaium/noexpensive/command/NoExpensiveCommand.kt
package cn.enaium.noexpensive.command

import net.minecraft.command.AbstractCommand
import net.minecraft.command.CommandSource
import net.minecraft.text.LiteralText
import net.minecraft.text.TranslatableText

/**
 * @author Enaium
 */
class NoExpensiveCommand : AbstractCommand() {
    override fun getCommandName(): String {
        return "noexpensive"
    }

    override fun getUsageTranslationKey(source: CommandSource): String? {
        return null
    }

    override fun execute(source: CommandSource, args: Array<out String>) {
        if (args.isEmpty()) {
            source.sendMessage(LiteralText("§cNoExpensive §7by §bEnaium"))
            return
        }
        if (args[0].equals("reload", ignoreCase = true)) {
            Config.load()
            source.sendMessage(TranslatableText("command.reload.success"))
        }
    }
}
